src/main/sensors/compass.c

```diff
@@ -70,6 +70,11 @@ bool compassInit(int16_t magDeclinationFromConfig)
     return ret;
 }
 
+bool isCompassHealthy(void)
+{
+    return (magADC[X] != 0) && (magADC[Y] != 0) && (magADC[Z] != 0);
```

Comment:
Not sure if any-zero or rather all-zero should indicate failure.
The code above is copied from imu.c